#!/usr/bin/env python3

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "dv_ros_imu_bias"
gen = ParameterGenerator()

# read_mode_enum = gen.enum([gen.const("Camera", int_t, 0, "Assumes there is a transformation available between the IMU and the camera."),
#                               gen.const("IMU", int_t, 1, "Assumes there is no transformation available between the IMU and the camera.")],
#                              "")
#
# gen.add("IMU_data_frame", int_t, 0,
#         "Set IMU data frame", 0, 0, 2,
#         edit_method=read_mode_enum)

gen.add("start_collecting", bool_t, 0, "Start collecting data for bias estimation.", False)

gen.add("variance_threshold", double_t, 0.0,
        "Set the Variance threshold.", 0.1, 0, 2)

gen.add("collection_duration", double_t, 0.0,
        "Set the collection duration [s].", 1.0, 1.0, 10)

exit(gen.generate(PACKAGE, "bias_estimation_node", "BiasesEstimation"))

# varianceThreshold
# gravityRange
# duration
